lib/msal-common/src/utils/UrlUtils.ts

```diff
@@ -9,6 +9,33 @@ import {
     createClientAuthError,
 } from "../error/ClientAuthError.js";
 import { StringDict } from "./MsalTypes.js";
+import { StringUtils } from "./StringUtils.js";
+
+/**
+ * Canonicalizes a URL by making it lowercase and ensuring it ends with /
+ * Inlined version of UrlString.canonicalizeUri to avoid circular dependency
+ * @param url - URL to canonicalize
+ * @returns Canonicalized URL
+ */
+function canonicalizeUrl(url: string): string {
```

Comment:
Don't we already have this? I think we do it specific to authority only: getCanonicalAuthority(). May be we can merge the functionality in the future.